src/license/src/manager.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::{LazyLock, RwLock};
+
+use jsonwebtoken::{Algorithm, DecodingKey, Validation};
+use serde::Deserialize;
+use thiserror::Error;
+
+/// License tier.
+///
+/// Each enterprise [`Feature`](super::Feature) is available for a specific tier and above.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Deserialize)]
```

Comment:
Notice the `PartialOrd` here. IIUC it assumes a full 1-dimension ordering among all tiers i.e. the feature set of Tier X must be a subset of Tier Y, and Tier Y must be a subset of Tier Z, etc. Does it sound good to you? @neverchanje

Reply:
Good to know. If we're going to introduce more tiers without total ordering on the feature set in the future, we can refactor the `for_all_features` macro to accept either a range or set of tiers, instead of specifying the min one.